app/models/piece.rb

```diff
@@ -20,46 +29,104 @@ def move_to!(new_x, new_y)
+    end
+    false
   end
 
-  def obstructed?(destination_x, destination_y)
-    # absolute catches negative coordinate conditions (when x and y go in opposite directions)
-    x_distance = (x_coordinate - destination_x).abs
-    y_distance = (y_coordinate - destination_y).abs
-
-    chess_in_between = []
-
-    # first, raise an argument error when self and sq have the same coordinates
-    raise ArgumentError, 'destination has to have a different location' if x_distance.zero? && y_distance.zero?
-
-    if x_distance.zero?
-      # the two locations are on the same vertical line
-      chess_in_between = game.pieces.where(x_coordinate: x_coordinate)
-                             .where(y_coordinate: y_coordinate + 1...destination_y)
-    elsif y_distance.zero?
-      # the two locations are on the same horizontal line
-      chess_in_between = game.pieces.where(y_coordinate: y_coordinate)
-                             .where(x_coordinate: x_coordinate + 1...destination_x)
-
-    elsif x_distance != y_distance
-      raise ArgumentError, "destination (#{destination_x}, #{destination_y}) must be on the diagonal of the origin (#{x_coordinate},#{y_coordinate})"
-    else
-      # TODO: coordinate + 1 but does not catch -1.
-      # the two locations are on the same diagonal line
-      chess_in_between = game.pieces.where(x_coordinate: x_coordinate + 1...destination_x)
-                             .where(y_coordinate: y_coordinate + 1...destination_y)
-                             .where("(x_coordinate - #{x_coordinate}) = (y_coordinate- #{y_coordinate})")
+  # checks for vertical obstruction
+  def obstructed_vertically?(y)
+    y_distance = (y_coordinate - y).abs
+
+    # if moving for vertical obstruction
+    y_dir = y > y_coordinate ? 1 : -1
+
+    # this checks if there are any steps in between the coordinate and destination
+    return false if y_distance < 2
+    # distance -1 means we go one less than the distance because we don't count the end distance
+    (1..(y_distance - 1)).each do |step| # number of steps vertically
+      x_steps = step * 0
+      y_steps = step * y_dir
+      x_square = x_coordinate + x_steps
+      y_square = y_coordinate + y_steps
+
+      return true if square_is_occupied(x_square, y_square)
     end
+    false
+  end
+
+  # check diagonal obstruction
+  def obstructed_diagonally?(x, y)
+    x_distance = (x_coordinate - x).abs
+    y_distance = (y_coordinate - y).abs
+
+    # if moving rt, x_dir is positive
+    x_dir = x > x_coordinate ? 1 : -1
+    # if moving up, y_dir is positive
+    y_dir = y > y_coordinate ? 1 : -1
+
+    # this checks if there are any steps in between the coordinate and destination
+    return false if (x_distance < 2) && (y_distance < 2)
+    # distance -1 means we go one less than the distance because we don't count the end distance
+    (1..(x_distance - 1)).each do |step|
+      x_steps = step * x_dir
+      y_steps = step * y_dir
```

Comment:
Figuring out to multiply by -1 to move backwards took a while.  This way x and y can go in different directions with the same loop.